### PR TITLE
[JBPM-9085],[JBPM-9086] - fixed failing tests

### PR DESCRIPTION
**Pull request by mareknovotny**
https://issues.redhat.com/browse/JBPM-9085

just in case adding check for null value

### CODE REVIEW

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by gmunozfe** [comment]
@mareknovotny  these failing tests weren't being executed because of BZ annotation. Are they needed now?

If yes, _testFilterParams_ test may be modified like this (line [147](https://github.com/kiegroup/jbpm/blob/master/jbpm-test-coverage/src/test/java/org/jbpm/test/functional/task/HumanTaskQueryFilterTest.java#L147)):
`QueryFilter queryFilter = new QueryFilter("t.id=1 AND t.priority=0", null, "t.name", true);`

The other one, _testSingleResult_ I think it's no longer supported, and could be deleted: https://bugzilla.redhat.com/show_bug.cgi?id=1132145
There's no "singleResult" param at QueryFilter

**Comment by MarianMacik** [comment]
Ah, so that's why it suddenly started failing :) I would then recommend this:

testSingleResult
* Feature was removed in https://github.com/kiegroup/jbpm/pull/690, so I believe we should delete also the test.

testFilterParams
* I believe that using both param names and param values in the first argument is abusing of the API, therefore we should change it to 

`QueryFilter queryFilter = new QueryFilter("t.id=:id AND t.priority=:priority", {Map with "id"->1, "priority"->0}, "t.name", true);`

What do you think?

**Comment by gmunozfe** [comment]
@MarianMacik I agree, filterParams are conditions with named parameters and map contains their values, so probably it has no sense to pass a null map, and therefore, null pointer exception is useful in this case (following Mario's tweet "There's no line of code deserving more thinking and explanations than a null check." ;-)

File: jbpm-test-coverage/src/test/java/org/jbpm/test/functional/task/HumanTaskQueryFilterTest.java [hunk 1]
```diff
@@ -123,28 +123,17 @@ public void testAscendingOrder() {
...
      * TODO - Need to pass in parameters which do make sense.
      */
     @Test
     @BZ("1132157")
     public void testFilterParams() {
         startHumanTaskProcess(10, "john's task", "john");
-
-        QueryFilter queryFilter = new QueryFilter("x=1,y=2", null, "t.name", true);
+        Map<String, Object> parameters = new HashMap<String, Object>();
+        parameters.put("id", 1);
```
Comment (gmunozfe) [inline jbpm-test-coverage/src/test/java/org/jbpm/test/functional/task/HumanTaskQueryFilterTest.java]: It should be "1L"

**Comment by mareknovotny** [comment]
@MarianMacik I would keep that null check there as that not replaced parameters will be even more clearer than just NPE thrown there. I am always fan of checks if it is not directly clear on first view or IAE should be thrown there instead.

**Comment by mareknovotny** [comment]
just in case you haven't noticed this failures start to shown over weekend or so. one of it is seanable at 
kiegroup/droolsjbpm-build-bootstrap/pull/12

**Comment by mareknovotny** [comment]
skip ci

**Comment by Ginxo** [comment]
tested locally and they are working. LGTM

**Pull request merged**